set.mm

```diff
@@ -699314,31 +699320,66 @@ number axioms (add ~ ax-10 , ~ ax-11 , ~ ax-13 , ~ ax-nul , and remove
-      JZUVHGZUVKWTZUWLFUWQUVKFUWQNODUWPFXAZFUWPUWSFXNZOUWPUWSFXCUWTXDUWPUWSFXEV
-      HSXFVDXGUWBUVJVJZUVKUWPVJUWRUWLNUWJUXAUWKLUVJXHXIXJUWBUVJPXKDUWPUVKUVHXLX
-      OXMXPDUVKUVKYFXQXRSUVGUUQFXSUVGUUQJKXTYAVDABUVDUVFUVMUVDUVAUUTUVBMIZUUTJM
-      IZUGUVFUVAUUTUVBJMYBUVMUVAUXBUXCUVFUVMUWDUVAUXBUVFNUWEUWDUVATZUXBUVBUUTMI
-      UUTUFZUVFUXDUUTUVBUXDUUQUXDUUQUXDUUPUXDUUPUVTUWAUVAYCZVTZYDVTZUXDUUPUXGUV
-      TUWAUVAYEYGZYHZUVBPRUXDYISYJUXDUUTUXJYKUXDUXEJUUQUFZKIUVFUXDJUUQUXDYLUXHU
-      XIYMUXDUXKUUPJKUXDUUPUUQUXGUXDUUQUUPUFUXDUUPUXFUXDUUPLUXFUXDYNUWDUVAYOYQU
-      UAUUBUUCYPYRUUDYSUVMUWDUVAWKZUXCUVFNUWEUWDUXLTZUXCUUTUVFUXMUUTUXMUUTUXMUU
-      QUVTUUQERUWAUXLUVTUUPUUPUUEZYDYTUXMUUPUVTUVSUWAUXLUXNYTUVTUWAUXLYEYGUUFVT
-      UUGUXMUUQUUPJKUXMUUPUVTUWAUXLYCUWDLUUPUUHUEUXLUWDLUUPUWDYNUVTUWAUUIUUJUUK
-      UULYPYRYSUUMUUNUUOXR $.
+      ovexd wf1o wf logf1o f1of mp1i eqid logdmss feqresmpt oveq2i dvlog eqtr3i
+      fveq2 oveq2 dvmptco ovif2 simpll abscld simplr absne0d reccld neg1cn 1cnd
+      mulcomd mulm1d divneg2d simpr ltled absnidd eqcomd negcon1ad oveq2d eqtrd
+      0red 3eqtrd sylanb recn ad2antrr rereccld mulridd cle simpl lenltd absidd
+      biimpar ifeqda eqtrid mpteq2ia eqtri ) EABAUAZUBUCZFUCZGHIZABJUULKIZUUKLU
+      DUEZJUFZJUGZMIZGZABJUUKKIZGUUNUUTNOADUULUURDUAZFUCZJUVBKIZEPUUMUUOUHUHBPQ
+      LUIIZUJZEEPUKZROULSPUVGROUMSOUUKBRZTZUOUVFUULUOLUNUPIZUVEUJZUVFUOUVJUVKUQ
+      UVJUVEURUSNUVKUVJNUVEUVJUSUVEUVJURUSNOQLUNQUTRZOVASLUTRZOVGSUNUTROVBSVCVD
+      VEUVJUVEVFVHVIUVJPVJUVKUVFVJLUNVKUVJPUVEVLVHVMUVIUUKUVHUUKPRZOUVHUUKUVHUU
+      KERZUUKLVNZUVHUUKELVOZUJZRUVOUVPTZBUVRUUKCVPUUKELVQVSZVRVTWAUVHUVPOUVHUVO
+      UVPUVTWBWAWCWDUURUHRUVIUUPUUQJJWEWFWJSOUVBUVFRZTZUVBUWAUVBPROUVBPUVEWGWAU
+      WAUVBLVNZOUWAUVBUVERZWKUWCUVBPUVEWHUWDUVBLUVBLNUWDLUVERZUVLUVMQLUDUEUWEVA
+      VGWIQLWLWMUVBLUVEWNWOWPWQWAWRUWBJUVBKXBEABUULGHIABUURGNOABCWSSPDUVFUVCGZH
+      IZDUVFUVDGZNOPFUVFWTZHIUWGUWHUWIUWFPHUWIUWFNODPUVQUJZFXAZUVFFUWJUWKFXCUWJ
+      UWKFXDOXEUWJUWKFXFXGUVFUWJVJOUVFUVFXHZXISXJVDXKDUVFUWLXLXMSUVBUULFXNUVBUU
+      LJKXOXPVDABUUSUVAUVHUUSUUPUUOUUQMIZUUOJMIZUGUVAUUPUUOUUQJMXQUVHUUPUWMUWNU
+      VAUVHUVSUUPUWMUVANUVTUVSUUPTZUWMUUQUUOMIUUOUFZUVAUWOUUOUUQUWOUULUWOUULUWO
+      UUKUWOUUKUVOUVPUUPXRZVTZXSVTZUWOUUKUWRUVOUVPUUPXTYAZYBZUUQPRUWOYCSYEUWOUU
+      OUXAYFUWOUWPJUULUFZKIUVAUWOJUULUWOYDUWSUWTYGUWOUXBUUKJKUWOUUKUULUWRUWOUUL
+      UUKUFUWOUUKUWQUWOUUKLUWQUWOYOUVSUUPYHYIYJYKYLYMYNYPYQUVHUVSUUPWKZUWNUVANU
+      VTUVSUXCTZUWNUUOUVAUXDUUOUXDUUOUXDUULUVOUULERUVPUXCUVOUUKUUKYRZXSYSUXDUUK
+      UVOUVNUVPUXCUXEYSUVOUVPUXCXTYAYTVTUUAUXDUULUUKJKUXDUUKUVOUVPUXCXRUVSLUUKU
+      UBUEUXCUVSLUUKUVSYOUVOUVPUUCUUDUUFUUEYMYNYQUUGUUHUUIUUJ $.
+  $}
+
+  ${
+    readvcot.d $e |- D = { y e. RR | ( sin ` y ) =/= 0 } $.
+    $( The support of sin ( ~ df-supp ) restricted to the reals is an open set.
+       (Contributed by SN, 7-Oct-2025.) $)
+    resuppsinopn $p |- D e. ( topGen ` ran (,) ) $=
```

Comment:
You could also have used ``( ( sin |` RR ) supp 0 )``, but this is fine too.

Reply:
It's a good idea. I tried to change it, but it meshes more inconveniently with ~ cnima, and theorems like ~ elsuppfnd use more hypotheses than the restricted abstraction does (~ elrab2).

I notice [smfpimne](https://us.metamath.org/mpeuni/smfpimne.html) could be useful but would require moving an entire section into main